app/models/grda_warehouse/cas_project_client_calculator/boston.rb

```diff
@@ -246,19 +246,39 @@ def most_recent_transfer_assessment_for_destination(client)
 
     # NOTE: this is also used in cohorts
     def days_homeless_in_last_three_years_cached(client)
-      days = most_recent_pathways_or_transfer(client)&.
-        question_matching_requirement('c_new_boston_homeless_nights_total')&.AssessmentAnswer
-      return days if days.present?
+      assessment = most_recent_pathways_or_transfer(client)
+      pre_calculated_days = client.processed_service_history&.days_homeless_last_three_years || 0
+      return pre_calculated_days unless assessment.present?
 
-      client.processed_service_history&.days_homeless_last_three_years
+      pathways_days_homeless(assessment, client).presence || pre_calculated_days
```

Comment:
Since lines 265 and 267 have `|| 0`, I don't think `pathways_days_homeless` can ever return `nil`? If it should, maybe change `pathways_days_homeless` to not default to `0` and instead of using `+`, do something like:
```ruby
days = [unsheltered_days, sheltered_days, warehouse_unsheltered_days, warehouse_sheltered_days].compact
return nil if days.blank?
days.sum
```

Reply:
Lines 269 and 270 also have `|| 0`.